### PR TITLE
TGUI Cyborg Docking Station

### PR DESCRIPTION
**Pull request by pgmzeta**
[SILICONS][UI][QOL]
## About the PR 
Converts the cyborg docking station interface to use TGUI. It splits the interface into separate tabs for the Occupant and Supplies. This maintains feature parity with the existing docking station interface to the best of my ability.

### Intended behavior changes
1. Clicking the repair button will automatically use the maximum amount of fuel or wire to repair
2. For cyborg conversion chambers, when someone is being converted the UI turns to a Syndicate theme
3. There's a dedicated eject button in the TGUI window
4. Instead of entering R/G/B as numbers for eye glow colour, it uses the default Byond color prompt (like paint colour)
5. Eyebots in docking stations show their power cell level and allow the rest of the interface to be manipulated
6. You can tell if a docking station is occupied by an AI-controlled cyborg or an independent cyborg

image

## Why's this needed? 
TGUI :)

## Changelog

```changelog
(u)glowbold
(*)Cyborg Docking Stations have a new interface.
```

### CODE REVIEW

**Comment by pgmzeta** [comment]
Set to draft; this is functional in-game, but I'm wagering I have work to do before it'll be mergable.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
nitpicks, looks great!

File: tgui/packages/tgui/interfaces/CyborgDockingStation/index.tsx [hunk 1]
```diff
@@ -13,14 +13,17 @@ import { CyborgDockingStationData } from './type';
...
 
+  if (data.conversion_chamber && data.occupant.kind === "human") theme = "syndicate";
+  if (!data.allow_current_user) theme = "noninteractive";
 
   return (
     <Window
       width={500}
       height={640}
       title="Cyborg Docking Station"
-      theme={data.occupant.kind === "human" && data.conversion_chamber ? "syndicate" : "neutral"}>
```
Comment (ZeWaka) [inline tgui/packages/tgui/interfaces/CyborgDockingStation/index.tsx]: why not just disable them?

Comment (pgmzeta) [reply tgui/packages/tgui/interfaces/CyborgDockingStation/index.tsx]: say more?

**Comment by pgmzeta** [comment]
Do we need `allow_clothes` anymore? It's only ever explicitly set to the default value (`1` i.e. `TRUE`) in unused maps.

**Comment by pgmzeta** [comment]
!merge_upstream

File: tgui/packages/tgui/interfaces/CyborgDockingStation/index.tsx (hunk 1 above)
Comment (pgmzeta) [reply tgui/packages/tgui/interfaces/CyborgDockingStation/index.tsx]: reworked disabling to put a notice up top, and to properly disable the buttons

**Comment by pgmzeta** [comment]
!merge_upstream

**Comment by pgmzeta** [comment]
!merge_upstream

**Pull request merged**